Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM criteo/mesosbuild:1.9.0-automake
+
+ENV MESOS_BUILD_DIR=/src/mesos
+ENV CONTAINER_LOGGER_DIR=/src/mesos-external-container-logger
+WORKDIR $MESOS_BUILD_DIR
+ADD . $CONTAINER_LOGGER_DIR
+
+RUN make DESTDIR=$CONTAINER_LOGGER_DIR/mesos-install install
+RUN mkdir build
+WORKDIR build
+RUN ln -s $MESOS_BUILD_DIR/3rdparty
+WORKDIR $CONTAINER_LOGGER_DIR
+RUN ln -s $MESOS_BUILD_DIR mesos
+RUN mkdir build
+WORKDIR build
+ENV CXXFLAGS="-isystem $MESOS_BUILD_DIR/3rdparty/protobuf-*/src -isystem $MESOS_BUILD_DIR/3rdparty/glog-*/src -isystem $MESOS_BUILD_DIR/3rdparty/boost-*[0-9] -isystem $CONTAINER_LOGGER_DIR/mesos-install/usr/include -isystem $CONTAINER_LOGGER_DIR/mesos-install/usr/lib64/mesos/3rdparty/usr/include"
+RUN cmake3 -DWITH_MESOS=$CONTAINER_LOGGER_DIR/mesos-install/usr/local/ -DMESOS_SRC_DIR=$CONTAINER_LOGGER_DIR/mesos ..
+RUN make -j 3
```

Comment:
Shouldn't we favor an image with just the tools and mesos lib/sources, and using it to build with docker run + volume? I think the later approach is more suited to iterative development.

Reply:
I agree, we don't care about the generated image in this case, we only need the .so. Having Container logger dir as a volume will avoid having to rebuild the image every time.

Reply:
I'm not sure to understand the suggestion, could you detail?

Reply:
Instead of ADD use VOLUME, instead of the last RUN use CMD that way we can build an image that is independent from the code of the container logger. Then when we want to build the .so we would do something like `docker build . -t container-logger-build && docker run container-logger-build -v $(pwd):/src/mesos-external-container-logger`